src/lib/Cropper.svelte

```diff
@@ -229,34 +233,57 @@
       restrictPosition
     )
 
-    dispatch('cropcomplete', {
+    oncropcomplete?.({
       percent: croppedAreaPercentages,
       pixels: croppedAreaPixels,
     })
   }
 
   // ------ Reactive statement ------
   //when aspect changes, we reset the cropperSize
-  $: if (imgEl) {
-    cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
-  }
+  $effect(() => {
+    if (imgEl) {
+      cropperSize = cropSize ? cropSize : helpers.getCropSize(imgEl.width, imgEl.height, aspect)
+    }
+  })
 
   // when zoom changes, we recompute the cropped area
-  $: zoom && emitCropData()
+  $effect(() => {
+    if (zoom) {
+      emitCropData()
+    }
+  })
 
-  $: if(aspect) {
-    computeSizes()
-    emitCropData()
-  } 
+  // this variable is required to prevent the effect into entering an infinite loop (https://github.com/ValentinH/svelte-easy-crop/pull/59#discussion_r1896254867)
+  let lastAspect = $state(aspect)
+  $effect(() => {
+    if (aspect !== lastAspect) {
+      lastAspect = aspect
+      computeSizes()
+      emitCropData()
+    }
+  })
```

Comment:
When do you reach the infinite loop? When changing the aspect prop? With the additional state, I expect this effect to run twice for every time the aspect prop changes. 🤔

Reply:
![image](https://github.com/user-attachments/assets/14519291-f2b8-4da4-b607-8e764a11b9f7)
This is what happens with only 
```ts
$effect(() => {
    if (aspect) {
      lastAspect = aspect
      computeSizes()
      emitCropData()
    }
  })
```

This is the loop warning:
![image](https://github.com/user-attachments/assets/8abd3c67-44c6-4b3a-bd8b-afc258bc68da)

Reply:
I think this is because of the `lastAspect` assignment. Why do we need it? This variable wasn't here before.

Reply:
![image](https://github.com/user-attachments/assets/a3866ee3-e9d4-45de-995d-61c82bdadd7f)
```ts
$effect(() => {
    if (aspect) {
      computeSizes()
      emitCropData()
    }
  })
```

same thing happens without it. Will try to find out why this is happening.

> I think this is because of the `lastAspect` assignment. Why do we need it? This variable wasn't here before.

I added it there to prevent the loop, it was the only fix I found

Reply:
Actually, do you know the use of this effect right here? It seems that the component works as it should without it.

Reply:
After testing for some while, it seems that this effect has no use, even when the aspect is dynamically updated.